common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -0,0 +1,818 @@
+#include <beerocks/bcl/network/network_utils.h>
+#include <beerocks/bcl/son/son_wireless_utils.h>
+
+#include <easylogging++.h>
+
+#include <cmath>
+
+//////////////////////////////////////////////////////////////////////////////
+////////////////////////// Local Module Definitions //////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+namespace bwl {
+namespace nl80211 {
+
+#define BUFFER_SIZE 4096
+#define CSA_EVENT_FILTERING_TIMEOUT_MS 1000
+
+// Allocate a char array wrapped in a shared_ptr
+#define ALLOC_SMART_BUFFER(size)                                                                   \
+    std::shared_ptr<char>(new char[size], [](char *obj) {                                          \
+        if (obj)                                                                                   \
+            delete[] obj;                                                                          \
+    })
+
+// Temporary storage for station capabilities
+struct SRadioCapabilitiesStrings {
+    std::string supported_rates;
+    std::string ht_cap;
+    std::string ht_mcs;
+    std::string vht_cap;
+    std::string vht_mcs;
+    std::string btm_supported;
+    std::string nr_enabled;
+    std::string non_pref_chan;
+    std::string cell_capa;
+};
+
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////// Local Module Functions ///////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+static ap_wlan_hal::Event nl80211_to_bwl_event(const std::string &opcode)
+{
+    if (opcode == "AP-ENABLED") {
+        return ap_wlan_hal::Event::AP_Enabled;
+    } else if (opcode == "AP-DISABLED") {
+        return ap_wlan_hal::Event::AP_Disabled;
+    } else if (opcode == "AP-STA-CONNECTED") {
+        return ap_wlan_hal::Event::STA_Connected;
+    } else if (opcode == "AP-STA-DISCONNECTED") {
+        return ap_wlan_hal::Event::STA_Disconnected;
+    } else if (opcode == "INTERFACE-ENABLED") {
+        return ap_wlan_hal::Event::Interface_Enabled;
+    } else if (opcode == "INTERFACE-DISABLED") {
+        return ap_wlan_hal::Event::Interface_Disabled;
+    } else if (opcode == "ACS-STARTED") {
+        return ap_wlan_hal::Event::ACS_Started;
+    } else if (opcode == "ACS-COMPLETED") {
+        return ap_wlan_hal::Event::ACS_Completed;
+    } else if (opcode == "ACS-FAILED") {
+        return ap_wlan_hal::Event::ACS_Failed;
+    } else if (opcode == "AP-CSA-FINISHED") {
+        return ap_wlan_hal::Event::CSA_Finished;
+    } else if (opcode == "BSS-TM-RESP") {
+        return ap_wlan_hal::Event::BSS_TM_Response;
+    } else if (opcode == "DFS-CAC-COMPLETED") {
+        return ap_wlan_hal::Event::DFS_CAC_Completed;
+    } else if (opcode == "DFS-NOP-FINISHED") {
+        return ap_wlan_hal::Event::DFS_NOP_Finished;
+    }
+
+    return ap_wlan_hal::Event::Invalid;
+}
+
+#if 0
```

Comment:
Why #if 0 ? Let's remove dead code.

Reply:
Ok, I am removing it. At most, we will add it back from dwpal/wav if needed.